src/services/refactors/convertExport.ts

```diff
@@ -258,4 +258,16 @@ namespace ts.refactor {
     function makeExportSpecifier(propertyName: string, name: string): ExportSpecifier {
         return factory.createExportSpecifier(/*isTypeOnly*/ false, propertyName === name ? undefined : factory.createIdentifier(propertyName), factory.createIdentifier(name));
     }
+
+    function getExportingModuleSymbol(node: Node, checker: TypeChecker) {
+        const parent = node.parent;
+        if (isSourceFile(parent)) {
+            return parent.symbol;
+        }
+        const symbol = parent.parent.symbol;
+        if (symbol.valueDeclaration && isExternalModuleAugmentation(symbol.valueDeclaration)) {
+            return checker.getMergedSymbol(symbol);
+        }
+        return symbol;

```

Comment:
I think you just always want the merged symbol. Otherwise you won't correctly catch when an existing module augmentation adds a default export (which admittedly is very weird).

```ts
// @Filename: foo.ts
/////*a*/export function bar() {}/*b*/;

// @Filename: augmenter.ts
////import "./foo";
////declare module "./foo" {
////    export default function foo(): void;
////}

// @Filename: /b.ts
////import { foo } from "./foo";
```

So maybe just

```suggestion
        const parent = node.parent;
        const symbol = checker.getMergedSymbol(isSourceFile(parent) ? parent.symbol : parent.parent.symbol);
        return symbol;
```

Reply:
Import in `/b.ts` will not be changed. Maybe need to use two `symbols`, something like this

```ts
const parent = node.parent;
const symbol = isSourceFile(parent) ? parent.symbol : parent.parent.symbol;
const mergedSymbol = checker.getMergedSymbol(symbol);
return symbol === mergedSymbol ? [symbol] : [symbol, mergedSymbol];
```

or handle merged symbols directly in `importTracker`

Reply:
(edited my last comment to *actually* use a default import)

```diff
- import { foo } from "./foo";
+ import foo from "./foo";

Reply:
I’m sorry but I can’t actually tell what issue is being discussed here but it seems to be the one that’s preventing this from being merged. Is there a concrete failing test we can discuss?

Reply:
Changes should be made to the `b.ts` and` c.ts` files - @DanielRosenwasser am I right? 

```ts
// @Filename: /a.ts
/////*a*/export function a() {}/*b*/;

// @Filename: /b.ts
////import "./a";
////declare module "./a" {
////    export default function a(): void;
////}

// @Filename: /c.ts
////import { a } from "./a";

goTo.select("a", "b");
edit.applyRefactor({
    refactorName: "Convert export",
    actionName: "Convert named export to default export",
    actionDescription: "Convert named export to default export",
    newContent: {
        "/a.ts":
`export default function a() {};`,
        "/b.ts":
`import "./a";
declare module "./a" {
    export default function a(): void;
}`,
        "/c.ts":
`import a from "./a";`
    }
});
```

If we use only the merged symbol, changes will only be made to the `/b.ts` file. I see two ways how we can fix it -
1. https://github.com/microsoft/TypeScript/pull/47829#discussion_r844880748,
2. https://github.com/microsoft/TypeScript/blob/95731f03979985609c9fd662f179be5b9cea3470/src/services/refactors/convertExport.ts#L167

handle merged symbols in the import tracker, I'm not sure how safe it is

Reply:
> Changes should be made to the `b.ts` and `c.ts` files

@a-tarasyuk that test shows no changes to `b.ts` 🤔

Reply:
@andrewbranch `default` is useless in `/b.ts`. I've changed that

Reply:
It’s actually not 100% clear to me what the expected behavior should be. This is a very weird edge case, and I think it could be unexpected to generate changes to a merged declaration of any export. Personally I would be fine with the refactor being unavailable here.